### PR TITLE
Initial details

### PR DESCRIPTION
**Pull request by sanovikov71**

### CODE REVIEW

**Comment by genaforvena** [comment]
Fix #8

File: app/src/main/java/org/imozerov/streetartview/IntentUtils.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.imozerov.streetartview;
+
+/**
+ * Created by sergei on 09.02.16.
+ */
+public class IntentUtils {
+
+    public static final String EXTRA_KEY_ART_OBJECT_DETAIL_ID = "EXTRA_KEY_ART_OBJECT_DETAIL_ID";
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/IntentUtils.java]: Put this into fragment/activity where it is needed.

File: app/src/main/java/org/imozerov/streetartview/IntentUtils.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.imozerov.streetartview;
+
+/**
+ * Created by sergei on 09.02.16.
+ */
+public class IntentUtils {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/IntentUtils.java]: Remove this class.

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.java [hunk 3]
```diff
@@ -2,17 +2,20 @@
...
+import io.realm.Realm;
+import io.realm.RealmList;
+import io.realm.RealmObject;
+
 import org.imozerov.streetartview.storage.model.RealmArtObject;
 import org.imozerov.streetartview.storage.model.RealmAuthor;
 import org.imozerov.streetartview.ui.model.ArtObjectUi;
 
+import rx.Observable;
+import rx.functions.Func1;
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.java]: Unused imports.

File: app/src/main/java/org/imozerov/streetartview/ui/detail/ArtObjectDetailOpener.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.imozerov.streetartview.ui.detail;
+
+/**
+ * Created by sergei on 10.02.16.
+ */
+public interface ArtObjectDetailOpener {
+    void openDetailActivity(String id);
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/ArtObjectDetailOpener.java]: IMHO 
`openArtObjectDetails(String id)`
will be better name. as activity is part of implementation but not the interface

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        compositeSubscription.unsubscribe();
+    }
+
+    private void startFetchingArtObject() {
+        this.compositeSubscription.add(
+                dataSource.getArtObject(getIntent().getStringExtra(IntentUtils.EXTRA_KEY_ART_OBJECT_DETAIL_ID))
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.java]: we better store id in field (also put in bundle etc) as this solution might not work in screen rotation case. Or I am wrong?

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    String artObjectId;
+
+    public ArtListViewHolder(Context aContext, View itemView) {
+        super(itemView);
+
+        ArtObjectDetailOpener detailOpener;
+        try {
+            detailOpener = (ArtObjectDetailOpener) aContext;
+        } catch (ClassCastException cce) {
+            throw new RuntimeException("Activity must implement ArtObjectDetailOpener interface");
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java]: Print the name of activity to be able to trace the problem faster.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ * Created by sergei on 09.02.16.
+ */
+public class ArtListViewHolder extends RecyclerView.ViewHolder {
+    TextView author;
+    TextView artObjectName;
+    CircleImageView thumb;
+
+    String artObjectId;
+
+    public ArtListViewHolder(Context aContext, View itemView) {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java]: remove a prefix

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java [hunk 8]
```diff
@@ -49,20 +47,7 @@ public int getItemCount() {
     }
 
     public void setData(List<ArtObjectUi> aArtObjectUis) {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java]: remove a prefix

File: app/src/main/java/org/imozerov/streetartview/IntentUtils.java (hunk 1 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/IntentUtils.java]: done

File: app/src/main/java/org/imozerov/streetartview/IntentUtils.java (hunk 2 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/IntentUtils.java]: done

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.java (hunk 3 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.java]: done

File: app/src/main/java/org/imozerov/streetartview/ui/detail/ArtObjectDetailOpener.java (hunk 4 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/ArtObjectDetailOpener.java]: done

File: app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.java (hunk 5 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/detail/DetailArtObjectActivity.java]: done

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java (hunk 6 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java]: done

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java (hunk 7 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListViewHolder.java]: done

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java (hunk 8 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java]: done

**Comment by genaforvena** [comment]
:+1:

**Pull request merged**